hyperspy/roi.py

```diff
@@ -83,9 +83,35 @@ def __init__(self):
             """, arguments=['roi'])
         self.signal_map = dict()
 
+    def __repr__(self):
+        attr = []
+        for name, value in self._get_attributes().items():
+            if value is t.Undefined:
+                attr.append(f"{name}={value}")
+            else:
+                # otherwise format value with the General specifer
+                attr.append(f"{name}={value:G}")
+        return f"{self.__class__.__name__}" + "(" + ", ".join(attr) + ")"
```

Comment:
To make it more readable:
```suggestion
        return f"{self.__class__.__name__}({', '.join(attr)})"
```